File: FestivalC#/Festival/Services/Validation/ValidatorTransaction.cs
using Festival.Model;
using Festival.Validation.Exceptions;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Festival.Validation
{
    public class ValidatorTransaction : IValidator<Transaction>
    {
        public void validate(Transaction transaction)
        {
            String errors = "";
            if (transaction.ClientName.Length == 0)
            {
                errors += "Client name must not be null.";
            }
            if (transaction.NumberOfTickets < 0)
            {
                errors += "Number of tickets must be positive.";
            }
            if (errors.Length != 0)
            {
                throw new ValidatorException(errors);
            }
        }
    }
}

File: FestivalJava/Services/src/main/java/festival/Validation/ValidatorTransaction.java
package festival.Validation;

import festival.Domain.Transaction;
import festival.Validation.Exceptions.ValidatorException;

/**
 * Created by Sebi on 18-Mar-17.
 */
public class ValidatorTransaction implements IValidator<Transaction> {
    @Override
    public void validate(Transaction transaction) throws ValidatorException {
        String errors = "";
        if (transaction.getClientName().length() == 0){
            errors += "Client name must not be null.";
        }
        if (transaction.getNumberOfTickets() < 0){
            errors += "Number of tickets must be positive.";
        }
        if (errors.length() != 0){
            throw new ValidatorException(errors);
        }
    }
}
